model/task_lifecycle.go

```diff
@@ -1402,10 +1402,13 @@ func updateBuildStatus(ctx context.Context, b *build.Build) (bool, error) {
 	buildStatus := getBuildStatus(buildTasks)
 	// If all the tasks are unscheduled, set active to false
 	if buildStatus.allTasksUnscheduled {
-		if err = b.SetActivated(ctx, false); err != nil {
-			return true, errors.Wrapf(err, "setting build '%s' as inactive", b.Id)
+		if b.Activated {
+			if err = b.SetActivated(ctx, false); err != nil {
+				return true, errors.Wrapf(err, "setting build '%s' as inactive", b.Id)
+			}
+			return true, nil
 		}
-		return true, nil
+		return false, nil
```

Comment:
If the build activation status wasn't changed in this case, is it still necessary to do any of the checks below this block, like checking for essential unfinished tasks? If not, then could we explain the reason why it's okay to return early here?

Reply:
It isn't possible for a build's `HasUnfinishedEssentialTask` field to change when all tasks are unscheduled, but it might be possible for the `AllTasksBlocked` field to have changed. Going to remove this line for that reason.